databases/week3/WEEK3 homework/mealSharing_db

```diff
@@ -0,0 +1,256 @@
+CREATE DATABASE Meal_Sharing;
+
+
+CREATE TABLE `Meal` (
+  `id` int UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY,
+  `title` varchar(255) NOT NULL,
+  `description` text,
+  `location` varchar(255) NOT NULL,
+  `Hosting_time` datetime NOT NULL,
+  `max_reservations` int NOT NULL,
+  `price` decimal NOT NULL,
+  `created_date` date
+)ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci;
+
+
+INSERT INTO Meal(title,description,location,Hosting_time,max_reservations,price,created_date)
+Values('crispy cheesy shrimp tacos','crispy cheesy shrimp tacos and homemade salsa naranja'     ,'Mexico City','2022-08-10 17:30:20',50,80,'2022-07-20'),
+       ('Handmade Pasta','Homemade tagliatelle with a simple ragu (sauce)-
+Panna cotta','Florence, Italy','2021-09-07 11:30:14',25,120,'2021-08-23'),
+		('paella','Traditional Valencian paella',' Almeria, Spain ','2021-02-24 18:20:45',
+        15,80,'2021-01-20'),
+        (' French Quiche','Autumn quiche and Spiced poached pears','Cologne, Germany','2022-05-31 20:30:00',10,75,'2022-06-15'),
+        (' South Indian Brunch','South Indian masala dosa, crispy, gluten-free rice+lentils crepe and Aloo masala, savory potato masala mash-A tangy cilantro coconut chutney','New York, USA','2022-05-14 11:30:30',25,90,'2022-04-12');
+
+
+-- Queries to write: Meal
+-- 1,  Get all meals
+SELECT *
+FROM Meal;
+
+-- Add a new meal
+INSERT INTO Meal(title,description,location,Hosting_time,max_reservations,price,created_date)
+VALUES('Vegetarian Tagine',NULL,'Marrakech, Morocco ','2022-03-17 17:30:00',25,50,'2022-02-10');
+ 
+ -- Get a meal with any id, fx 1
+SELECT *
+FROM Meal
+WHERE id = 3;
+
+-- Update a meal with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE Meal 
+SET title = 'Paella',description=' Spanish classic, the Valencian paella'
+WHERE title='paella';
+
+-- Delete a meal with any id, fx 1
+DELETE
+FROM Meal
+WHERE price >100;
+
+-- Reservation
+
+CREATE TABLE `Reservation` (
+  `id` int UNSIGNED NOT NULL AUTO_INCREMENT PRIMARY KEY  ,
+  `number_of_guests` int  NOT NULL ,
+  `meal_id` int UNSIGNED NOT NULL,
+  `created_date` date,
+  `contact_phonenumber` varchar(255),
+  `contact_name` varchar(255),
+  `contact_email` varchar(255),
+    CONSTRAINT `fk_Meal` FOREIGN KEY (`meal_id`) REFERENCES `Meal` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
```

Comment:
If we delete all reservations for a given meal, the meal entry will also be deleted, is that what you want? (It is ok, just checking )

Reply:
no I don't want the changes in other table affect meal table. shall I remove on delete cascade??

Reply:
You might want to use `ON DELETE SET NULL`, if the meal_id in that table can be NULL. If you remove the `ON DELETE CASCADE` completely the default is `ON DELETE RESTRICT` and then you cannot delete a meal in the meal table that is referenced in this table.
Does it make sense?

Reply:
yes , so if we delete one meal , we can not delete the rows which are referenced to that meal , we can only set the meal id to null. (That is how I got it.)

Reply:
Let me give you an example for the 3 options:
 - with "ON DELETE SET NULL" when you delete a meal with id=2, all reservations for meal_id=2 will have the meal_id set to NULL automatically
 -with "ON DELETE CASCADE" when you delete a meal with id=2, all reservations for meal_id=2 will  be deleted automatically
 -with "ON DELETE RESTRICT" you can only delete a meal with id=2 if there are no reservations with meal_id=2